nautobot/extras/signals.py

```diff
@@ -127,7 +127,7 @@ def _handle_changed_object(sender, instance, raw=False, **kwargs):
 
     # Housekeeping: 0.1% chance of clearing out expired ObjectChanges
     changelog_retention = get_settings_or_config("CHANGELOG_RETENTION")
-    if changelog_retention and random.randint(1, 1000) == 1:
+    if changelog_retention and secrets.randbelow(1000) == 0:
```

Comment:
Looks like the performance impact is minimal (default for timeit is 10k iterations)

```py
>>> timeit.timeit(setup="import random", stmt="random.randint(1, 1000)")
1.0842244539999228
>>> timeit.timeit(setup="import secrets", stmt="secrets.randbelow(1000)")
2.6048511999979382
>>> timeit.timeit(setup="import random", stmt="random.randint(1, 1000)")
1.0478729710011976
>>> timeit.timeit(setup="import secrets", stmt="secrets.randbelow(1000)")
1.6280573209987779
```

Reply:
Thank you for checking me on this!

Reply:
Hopefully this particular code is not long for this world in any case... :-)